tests/test_create_index_seaice.F90

```diff
@@ -0,0 +1,175 @@
+
+     ! Parse the index info in cbuf, and fill gfld with the info about
+     ! the first message.
+     jdisc = -1
+     do i = 1, 13
+        jids(i) = -9999
+     end do
+     jpdtn = -1
+     do i = 1, 100
+        jpdt(i) = -9999
+     end do
+     jgdtn = -1
+     do i = 1, 250
+        jgdt(i) = -9999
+     end do
+     do j = 1, nnum
+        print *, '   testing unpacking index buffer with getgb2s2() for message', j
+        call getgb2s2(cbuf, idxver, nlen, nnum, j - 1, jdisc, jids, jpdtn, jpdt, jgdtn, &
+             jgdt, k, gfld, lpos, iret)
+        if (iret .ne. 0) stop 101
+
+        ! print *, '****************'
+        ! print *, gfld%version, gfld%discipline, gfld%idsectlen, gfld%ifldnum, gfld%griddef
+        ! print *, gfld%ngrdpts, gfld%numoct_opt, gfld%interp_opt, gfld%num_opt
+        ! print *, gfld%igdtnum, gfld%igdtlen
+        ! print *, gfld%ipdtnum, gfld%ipdtlen, gfld%num_coord
+        ! print *, gfld%unpacked, gfld%ibmap
+        ! print *, 'sec1:', gfld%idsect
+        ! print *, 'gdt:', gfld%igdtmpl
+
+        ! Check that the information is correct for the first record.
+        if (gfld%version .ne. 2) stop 102
+        if (gfld%discipline .ne. 10) stop 103
+        if (gfld%idsectlen .ne. 13) stop 110
+        if (gfld%ifldnum .ne. 1) stop 111
+        if (gfld%griddef .ne. 0) stop 112
+        if (gfld%ngrdpts .ne. 259200) stop 120
+        if (gfld%numoct_opt .ne. 0 .or. gfld%interp_opt .ne. 0 .or. gfld%num_opt .ne. 0) stop 122
+        if (gfld%igdtnum .ne. 0 .or. gfld%igdtlen .ne. 19) stop 123
+        if (gfld%ipdtnum .ne. 0 .or. gfld%ipdtlen .ne. 15 .or. gfld%num_coord .ne. 0) stop 130
+        if (gfld%unpacked .neqv. .FALSE.) stop 131
+        if (gfld%ibmap .ne. 255) stop 132
+        do i = 1, gfld%idsectlen
+           if (gfld%idsect(i) .ne. expected_idsect(i)) then
+              print *, gfld%idsect
+              stop 200
+           endif
+        end do
+        do i = 1, gfld%igdtlen
+           if (gfld%igdtmpl(i) .ne. expected_igdtmpl(i)) then
+              print *, gfld%igdtmpl
+              stop 210
+           endif
+        end do
+        do i = 1, gfld%ipdtlen
+           if (gfld%ipdtmpl(i) .ne. expected_ipdtmpl(i)) then
+              print *, i, gfld%ipdtmpl(i), expected_ipdtmpl(i)
+              print *, 'gfld%ipdtmpl', gfld%ipdtmpl
+              print *, 'expected_ipdtmpl', expected_ipdtmpl
+              stop 220
+           endif
+        end do
+        do i = 1, gfld%idrtlen
+           if (gfld%idrtmpl(i) .ne. expected_idrtmpl(i)) then
+              print *, i, gfld%idrtmpl(i), expected_idrtmpl(i)
+              print *, 'gfld%idrtmpl', gfld%idrtmpl
+              print *, 'expected_idrtmpl', expected_idrtmpl
+              stop 230
+           endif
+        end do
+
+        ! Free memory.
+        call gf_free(gfld)
+        print *, '   OK!'
+     end do
+
+     ! Clean up.
+     deallocate(cbuf)
+     call gf_finalize(iret)
+     if (iret .ne. 0) stop 200
```

Comment:
Would it make sense to change this to something other than `stop 200` since that number is already used?  Otherwise looks good.

Reply:
Sure, but I really only track that down if I start having trouble with the test.

That is, if the test starts failing, and I need them, then I will fix up the stop numbers. But a repeated number is not a bug because it is still enough to cause the test to fail.

So you don't have to worry about the stop numbers in test codes when you inspect them. Just make sure a non-zero stop happens when there is a problem.